pkg/controller/certificates/issuing/internal/keystore.go

```diff
@@ -59,17 +60,37 @@ func encodePKCS12Keystore(password string, rawKey []byte, certPem []byte, caPem
 	if len(certs) > 1 {
 		cas = append(certs[1:], cas...)
 	}
-	return pkcs12.LegacyRC2.Encode(key, certs[0], cas, password)
+
+	switch algorithm {
+	case cmapi.AESPKCS12Algorithm:
+		return pkcs12.Modern2023.Encode(key, certs[0], cas, password)
+	case cmapi.DES3PKCS12Algorithm:
+		return pkcs12.LegacyDES.Encode(key, certs[0], cas, password)
+	case cmapi.RC2PKCS12Algorithm:
+		return pkcs12.LegacyRC2.Encode(key, certs[0], cas, password)
+	default:
+		return pkcs12.LegacyRC2.Encode(key, certs[0], cas, password)
+	}
```

Comment:
suggestion: Specifying `AES256` doesn't specify the key derivation function (KDF) or HMAC algorithm. `AES256` (or `DES3` or `RC2`) isn't enough to identify what's actually happening, and if the underlying library changed to use a different KDF or HMAC algorithm that would be a breaking change for users.

I'll add a suggested fix in a different comment